activitypub/database/database.go

```diff
@@ -18,22 +24,48 @@ import (
 )
 
 var (
-	regexpID     = regexp.MustCompile("^https://(.+)/([^/]+)$")
-	regexpOutbox = regexp.MustCompile("^https://(.+)/outbox$")
-	regexpInbox  = regexp.MustCompile("^https://(.+)/inbox$")
+	regexpID        = regexp.MustCompile("/user/([^\\/]+)$")
+	regexpOutbox    = regexp.MustCompile("/user/([^\\/]+)/outbox$")
+	regexpInbox     = regexp.MustCompile("/user/([^\\/]+)/inbox$")
+	regexpRead      = regexp.MustCompile("/user/([^\\/]+)/read/(.*)")
+	regexpFollowers = regexp.MustCompile("/user/([^\\/]+)/followers")
+)
```

Comment:
Lack of anchors may be concerning?

Reply:
Hm. The string that's matched is the ID of the object, which is a full url, so I can't anchor to the start. We might eventually want to have an `ID` type or something that we can extract the host/path/etc from, but that's out of scope here. I'll add ending anchors for now.